app/javascript/elm/src/Data/Overlay.elm

```diff
@@ -0,0 +1,76 @@
+module Data.Overlay exposing (Model, Operation(..), Overlay(..), init, none, update, view)
+
+import Html exposing (Html, div, p, text)
+import Html.Attributes exposing (class, id)
+
+
+type Model
+    = Overlays (List Overlay)
+
+
+type Overlay
+    = HttpingOverlay
+    | IndoorOverlay
+    | PopupOverlay
+    | TimeFrameOverlay
+
+
+type Operation
+    = AddOverlay Overlay
+    | RemoveOverlay Overlay
+
+
+none : Model
+none =
+    Overlays []
+
+
+init : Bool -> Model
+init isIndoor =
+    if isIndoor then
+        Overlays [ IndoorOverlay ]
+
+    else
+        none
+
+
+update : Operation -> Model -> Model
+update operation model =
+    case ( operation, model ) of
+        ( AddOverlay overlay, Overlays [] ) ->
+            Overlays [ overlay ]
+
+        ( AddOverlay overlay, Overlays overlays ) ->
+            Overlays <| overlay :: overlays
+
+        ( RemoveOverlay overlay, Overlays [] ) ->
+            none
+
+        ( RemoveOverlay overlay, Overlays overlays ) ->
+            Overlays <| List.filter ((/=) overlay) overlays
+
+
+view : Model -> Html msg
+view (Overlays overlays) =
+    case List.head overlays of
+        Just overlay ->
+            case overlay of
+                HttpingOverlay ->
```

Comment:
I'd probably go with `Loading`

Reply:
I like your suggestion better but for now I'm leaving it the way it is cause the verb httping is used in multiple places and prefer consistency now. I'm not sure how I would rename the rest